### PR TITLE
Created buffer cursor.

### PR DESCRIPTION
**Pull request by Hlebanja**
General improvements to listing.
Created tests in test-worker for listing collection.

### CODE REVIEW

File: Database/Archive.ts [hunk 1]
```diff
@@ -70,20 +70,21 @@ export class Archive<T = any> extends Silo<T, Archive<T>> {
...
 		} = []
 		let limit = cursor?.limit ?? Selection.standardLimit
 		let newCursor: string | undefined
-		for (const prefix of Cursor.prefix(cursor)) {
+		const prefixes = Cursor.Archive.prefix(cursor)
+		for (const prefix of prefixes) {
 			const loaded = await this.backend.doc.list({
 				prefix: this.partitions + prefix,
-				limit,
+				limit: limit,
```
Comment (nilssonemma) [inline Database/Archive.ts:87]:
```suggestion
				limit,
```

File: Database/Configuration/Buffer.spec.ts [hunk 2]
```diff
@@ -1,5 +1,14 @@
+import * as cryptly from "cryptly"
 import { Buffer } from "./Buffer"
 
+function createIds(): string[] {
+	const mask = 255
+	const result: string[] = []
+	for (let seed = 0; seed < mask + 1; seed++) {
+		result.push(cryptly.Identifier.fromBinary(new Uint8Array([seed ?? 0])) + "AA")
+	}
```
Comment (nilssonemma) [inline Database/Configuration/Buffer.spec.ts:9]:
```suggestion
	for (let seed = 0; seed < mask + 1; seed++)
		result.push(cryptly.Identifier.fromBinary(new Uint8Array([seed ?? 0])) + "AA")
```

File: Database/Cursor/Buffer/index.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+		return base ? { ...base, shard: parsed?.shard } : undefined
+	}
+	export function from(selection: Selection | any): Buffer | undefined {
+		let result: Buffer | undefined
+		const base = Base.from(selection)
+		if (selection?.cursor)
+			result = base?.shard ? base : undefined
+		else
+			result = base
+		return result
```
Comment (nilssonemma) [inline Database/Cursor/Buffer/index.ts:21]:
```suggestion
		const base = Base.from(selection)
		return selection?.cursor ? (base?.shard ? base : undefined) : base
```

File: test-worker/Context/TestSetup.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+				.map(user => ({ ...user, created: isoly.Date.previous(lastDate, date) + createdTime }))
+			result.push(...toBeAdded)
+		}
+		return result
+	}
+	export function createIds(ids: number, suffix?: string): string[] {
+		const result: string[] = []
+		for (let seed = 0; seed < ids; seed++) {
+			result.push(cryptly.Identifier.fromBinary(new Uint8Array([seed ?? 0])) + suffix ?? "AA")
+		}
```
Comment (nilssonemma) [inline test-worker/Context/TestSetup.ts:39]:
```suggestion
		for (let seed = 0; seed < ids; seed++)
			result.push(cryptly.Identifier.fromBinary(new Uint8Array([seed ?? 0])) + suffix ?? "AA")
```

**Review by nilssonemma — CHANGES_REQUESTED** [review]

File: test-worker/db/collection/user/create.spec.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+const users: model.User[] = [
+	user,
+	{ ...user, id: "AAAB" },
+	{ ...user, id: "AAAC" },
+	{ ...user, id: "aaaD" },
+	{ ...user, id: "aaaE" },
+	{ ...user, id: "aaaF" },
+	{ ...user, id: "aaaG" },
+]
```
Comment (marcushanses) [inline test-worker/db/collection/user/create.spec.ts:83]: Remove this file.

**Review by marcushanses — CHANGES_REQUESTED** [review]

File: test-worker/db/collection/user/create.spec.ts (hunk 5 above)
Comment (Hlebanja) [reply test-worker/db/collection/user/create.spec.ts]: Ops. Nice catch.

**Review by marcushanses — APPROVED** [review]

**Review by nilssonemma — APPROVED** [review]

**Pull request merged**